src/ssat/core/SsatSolver.cc

```diff
@@ -143,6 +143,9 @@ SsatSolver::parse_SDIMACS( gzFile & input_stream )
        else if ( *in == 'r' ) { // r p x_i  <-- randomly quantified variable x_i w.p. p 
           ++in;
           parsed_prob = parseFloat(in);
+          if (parsed_prob > 1.0) {
```

Comment:
Do we also parse negative numbers?

It seems we don't so here we only check `parsed_prob > 1.0` instead of `parsed_prob > 1.0 || parsed_prob < 0`.

However, the latter is more consistent with the error message.